include/wil/safearrays.h

```diff
@@ -0,0 +1,897 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_SAFEARRAYS_INCLUDED
+#define __WIL_SAFEARRAYS_INCLUDED
+
+#ifdef _KERNEL_MODE
+#error This header is not supported in kernel-mode.
+#endif
+
+#include <new.h> // new(std::nothrow)
+#include "wistd_type_traits.h"
+#include "resource.h" // unique_hkey
+
+namespace wil
+{
+#if defined( _OLEAUTO_H_ ) && !defined(__WIL_SAFEARRAY_)

```

Comment:
Given that this is a dedicated header file, the `_OLEAUTO_H_` check seems unnecessary and the file should include its dependencies

Reply:
It didn't used to be a dedicated header, so the transition may be incomplete.  I just looked at `filesystem.h` and it includes `PathCch.h` and `registry.h` includes `winreg.h`, so I guess it's okay to pull include `oleauto.h`.